Deprecation warnings

From current builds…

```
$ ( pylint --output-format=colorized --disable=R,C spinn_front_end_common; exit $(($? & 35)) )
************* Module spinn_front_end_common.interface.interface_functions.placements_provenance_gatherer
spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py:49:23: W0703: Catching too general exception Exception (broad-except)
spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py:52:12: W1505: Using deprecated method warn() (deprecated-method)
spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py:54:16: W1505: Using deprecated method warn() (deprecated-method)
************* Module spinn_front_end_common.utilities.utility_objs.extra_monitor_scp_processes.read_status_process
spinn_front_end_common/utilities/utility_objs/extra_monitor_scp_processes/read_status_process.py:74:12: W1505: Using deprecated method warn() (deprecated-method)
```

In particular, there are three places where we are using `warn()` when we should use `warning()` instead.

--- a/spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py
+++ b/spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py
@@ -14,12 +14,13 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import logging
+import traceback
+from spinn_utilities.log import FormatAdapter
 from spinn_utilities.progress_bar import ProgressBar
 from spinn_front_end_common.interface.provenance import (
     AbstractProvidesProvenanceDataFromMachine)
-import traceback
 
-logger = logging.getLogger(__name__)
+logger = FormatAdapter(logging.getLogger(__name__))
 
 
 class PlacementsProvenanceGatherer(object):
@@ -46,11 +47,11 @@ class PlacementsProvenanceGatherer(object):
                     prov_items.extend(
                         placement.vertex.get_provenance_data_from_machine(
                             transceiver, placement))
-                except Exception:
+                except Exception:  #pylint: disable=broad-except
                     errors.append(traceback.format_exc())
         if errors:
-            logger.warn("Errors found during provenance gathering:")
+            logger.warning("Errors found during provenance gathering:")
             for error in errors:
-                logger.warn(error)
+                logger.warning("{}", error)
 
         return prov_items

--- a/spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py
+++ b/spinn_front_end_common/interface/interface_functions/placements_provenance_gatherer.py
@@ -47,7 +47,7 @@ class PlacementsProvenanceGatherer(object):
                     prov_items.extend(
                         placement.vertex.get_provenance_data_from_machine(
                             transceiver, placement))
-                except Exception:  #pylint: disable=broad-except
+                except Exception:  # pylint: disable=broad-except
                     errors.append(traceback.format_exc())
         if errors:
             logger.warning("Errors found during provenance gathering:")

--- a/spinn_front_end_common/utilities/utility_objs/extra_monitor_scp_processes/read_status_process.py
+++ b/spinn_front_end_common/utilities/utility_objs/extra_monitor_scp_processes/read_status_process.py
@@ -71,7 +71,7 @@ class ReadStatusProcess(AbstractMultiConnectionProcess):
                     callback=self.__handle_reinjection_status_response)
         self._finish()
         if self.is_error():
-            logger.warn("Error(s) reading reinjection status:")
+            logger.warning("Error(s) reading reinjection status:")
             for (e, tb) in zip(self._exceptions, self._tracebacks):
                 traceback.print_exception(type(e), e, tb)
         return self._reinjection_status

